activerecord/lib/active_record/attributes.rb

```diff
@@ -259,6 +259,10 @@ def load_schema! # :nodoc:
         NO_DEFAULT_PROVIDED = Object.new # :nodoc:
         private_constant :NO_DEFAULT_PROVIDED
 
+        def adapter_name(connection)
+          connection.adapter_name.downcase.to_sym
```

Comment:
This keeps the parity with this method: 

https://github.com/rails/rails/blob/master/activerecord/lib/active_record/type.rb#L52